embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/clock/Clock.kt

```diff
@@ -20,11 +20,25 @@ public fun interface Clock {
 /**
  * Turns a number that specifies a millisecond value to nanoseconds
  */
-@InternalApi
-public fun Long.millisToNanos(): Long = TimeUnit.MILLISECONDS.toNanos(this)
+internal fun Long.millisToNanos(): Long = TimeUnit.MILLISECONDS.toNanos(this)
 
 /**
  * Turns a number that specifies a nanosecond value to milliseconds
  */
-@InternalApi
-public fun Long.nanosToMillis(): Long = TimeUnit.NANOSECONDS.toMillis(this)
+internal fun Long.nanosToMillis(): Long = TimeUnit.NANOSECONDS.toMillis(this)
+
+/**
+ * Any epoch timestamp that we detect to be unreasonable to be interpreted as milliseconds, we assume it's an unintended use of nanoseconds
+ * based on an old API version or assumption from OpenTelemetry conventions
+ */
+internal fun Long.normalizeTimestampAsMillis(): Long =
+    if (this < MAX_MS_CUTOFF) {
+        this
+    } else {
+        this.nanosToMillis()
```

Comment:
Can we log an error/warning or something so there's a chance a dev might spot this problem when developing locally? I am not usually a fan of this type of magical behavior but agree it's probably the best solution for this problem...

Reply:
Ah yea. Meant to do this in a separate PR. Will do so today,

And for the record, I also don't like this - but to maintain backwards compat at least in the short term, I was OK with it. Once we detect that folks aren't doing this anymore (via the property), we'll remove this